CoreWordCloud/src/au/gov/asd/tac/constellation/views/wordcloud/phraseanalysis/PhraseAnalysisModelLoader.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2010-2023 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.wordcloud.phraseanalysis;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author twilight_sparkle
+ */
+public class PhraseAnalysisModelLoader {
+
+    private static final Map<String, Set> excludedWords = new HashMap<>();
+    private static final Map<String, Set> delimiters = new HashMap<>();
+
+    private static final String COMMENT_TOKEN = "##";
+    private static final String SET_TOKEN = "#{";
+    private static final String UNICODE_RANGE_TOKEN = "#|";
+    private static final String OPEN_SET = "{";
+    private static final String CLOSE_SET = "}";
+
+    private static boolean isLoaded = false;
+    
+    private PhraseAnalysisModelLoader() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static Map<String, Set> getExcludedWords() {
+        return new HashMap<>(excludedWords);
+    }
+
+    public static Map<String, Set> getDelimiters() {
+        return new HashMap<>(delimiters);
+    }
+    
+    public static void loadMap() throws IOException {
+
+        if (isLoaded) {
+            return;
+        }
+
+        final BufferedReader excludedWordsReader = new BufferedReader(new InputStreamReader(PhraseAnalysisModelLoader.class.getResourceAsStream("ExcludedWords.u8"), StandardCharsets.UTF_8.name()));
+        processLines(excludedWordsReader, excludedWords, false);
+
+        final BufferedReader delimitersReader = new BufferedReader(new InputStreamReader(PhraseAnalysisModelLoader.class.getResourceAsStream("Delimiters.u8"), StandardCharsets.UTF_8.name()));
+        processLines(delimitersReader, delimiters, true);
+
+        isLoaded = true;
+    }
+
+    private enum MODE {
+        ADD_CHARACTERS, ADD_SET_OF_SETS, ADD_UNICODE_RANGE, LOOK_FOR_MODE_TOKEN
+    }
+
+    private static void processLines(final BufferedReader reader, final Map<String, Set> map, final boolean singleCharacterLines) throws IOException {
+
+        MODE currentMode = MODE.LOOK_FOR_MODE_TOKEN;
+        Set currentSet = new HashSet<>();
```

Comment:
can a type be set here rather than just using the raw type?

Reply:
As this set has Strings, chars and int[] added to it depending on the switch cases below, I don't think so

Reply:
Does it have multiple types at once or is it always the same?
The former could be fixed with Set<Object> (or ? extends Object perhaps) and the latter could be fixed with Set<?> (I think) although note that while better than using the rawtype, they're probably not much better.

Possibly in the future should look at how it can written such that the types can be a little more assured

Reply:
Neither <?> or <? extends Object> work with the current implementation.

Reply:
Looks like Set<Object> works having done a local check. That said, might still be good to find an option that doesn't require being so generic